app.py

```diff
@@ -1,2 +1,57 @@
 from flask import Flask, request
 import json
+
+
+app = Flask(__name__)
+with open("restaurants.json", "r") as read_file:
+    data = json.load(read_file)
+
+restaurants = data['restaurants']
+
+
+@app.route('/api/restaurants', methods=['GET', 'POST'])
+def all_restaurants():
+    if request.method == 'POST':
+        content = request.get_json(silent=True)
+        print(type(content))
+        if content is not None:
+            print(content)
+            restaurants.append(content['restaurant_name'])
+        response = app.response_class(
+            response=json.dumps({'restaurants': restaurants}),
+            status=201,
+            mimetype='application/json'
+        )
+        return response
```

Comment:
Flask has a nice pattern when you want to return json response:
```python
from flask import jsonify

@app.route("/api/test")
def route():
   ....
   return jsonify(my_object), status_code # e.g. jsonify({'restaurants': restaurants}), 201
```

Reply:
fixed